### PR TITLE
Bjorncs/policy description

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

File: security-utils/src/main/java/com/yahoo/security/tls/policy/PeerPolicy.java [hunk 1]
```diff
@@ -48,12 +59,18 @@ public boolean equals(Object o) {
...
 
     @Override
     public int hashCode() {
-        return Objects.hash(policyName, assumedRoles, requiredCredentials);
+        return Objects.hash(policyName, description, assumedRoles, requiredCredentials);
+    }
+
+    @Override
+    public int compareTo(PeerPolicy o) {
+        return policyName.compareTo(o.policyName);
```
Comment (vekterli) [inline security-utils/src/main/java/com/yahoo/security/tls/policy/PeerPolicy.java:74]: Do we enforce policy name presence and/or uniqueness today? If not, what happens if someone adds multiple policies without setting names? Should this be a multi-dimensional comparison if this is an issue?

**Review by vekterli — COMMENTED** [review]
Quick question about `compareTo` vs `equals` semantics to resolve first, otherwise LGTM

File: security-utils/src/main/java/com/yahoo/security/tls/policy/PeerPolicy.java (hunk 1 above)
Comment (bjorncs) [reply security-utils/src/main/java/com/yahoo/security/tls/policy/PeerPolicy.java:74]: Both are enforced in the constructor of `AuthorizedPeer`.
I just discovered that I have to rewrite `compareTo` to be consistent with `equals`. `TreeSet.equals` is currently reporting false positives 😨

Comment (bjorncs) [reply security-utils/src/main/java/com/yahoo/security/tls/policy/PeerPolicy.java]: There is no simple way to implement `Comparable` to be "consistent with equals" (ref Javadoc for `Comparable`). I have removed the commit - it was really not necessary (only intended to improve assertion error messages in a related unit test).

**Comment by bjorncs** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]

**Pull request merged**